README.md

```diff
@@ -8,16 +8,165 @@ Local.
+    - [isActual](#isactual)
+    - [example code](#example-code)
+- [Discovery](#discovery)
+- [Query](#query)
+- [Setup Cloudant instance](#setup-cloudant-instance)
+- [Testing](#testing)
+- [More Info](#more-info)
+- [Feature backlog](#feature-backlog)
+    - [View](#view)
+    - [Index](#index)
+
+<!-- /TOC -->
+
+# Design
+Loopback tries best to fit its model to a specific database's design, while limited by the nature of database, it's not always possible to support all loopback features perfectly, and user would like to be aware of some key features about cloudant before they start to design a cloudant model.
+
+## No partial update
+
+*Cloudant does not support the idea of updating a document. All "updates" on a document are _destructive_ replacements.*
 
-## Key features
+For example:
+
+```
+// original document
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "1",
+  "prop2": "2",
+}
+
+// data to be updated
+ds.updateOrCreate('User', {
+  prop1: 'updated1',
+}, function (err, res) {});
+
+// document after update
+{
+  "id": ...,
+  "_rev": ...,
+  "prop1": "updated1",
+}
+```
+
+*Please note how property `prop2` was completely dropped upon update.*
+
+Workaround: Do not pass partial values for the data object to be updated. If there are properties that are not being updated, please include the old value to keep data persistent.
+
+We plan to support partial update in the future by retrieving the original data and merging it with the updated one, but user should be aware of the risk when merging happens.
+
+## No frequent modification
+
+Cloudant is not designed to change same document frequently and multiple times. It stores status change by creating different documents and including the same unique id to tell that they are attached to the same item, not updating the same document.
+
+By modeling the data in separate documents that are only written once, we can ensure that documents never become conflicted, such as during concurrent access to the same document by separate processes.
+
+## Conflict control
+
+TBD. 
+
+The basic idea is when modifying a document, user needs to control conflict by handling the revision of a document, currently the connector controls this process, after retriving the latest revision, connector uses it to update/delete doc, and returns 409 conflict error if doc changes during that time slot. In the middle user could not interfere and provide their own conflict solution.
+
+This part will be done after we complete the `_rev` property refactor.
+
+# Model 
+
+## Map between model and document
+
+Unlike relational db or mongodb, cloudant doesn't have a concept as 'table' or 'collection', data in a cloudant database are all stored as documents.
+
+The connector uses a design document to represent a loopback model, and other non-design(non-design might not be the best word here, appreciate more ideas about how to describe it) documents to represent model instances.
```

Comment:
@0candy Let me elaborate more on the document type, there are two types of documents in cloudant database:
- Design document
- common document that is not a design document

I am not sure what is the brief and accurate word to describe the second one...